ofrak_patch_maker/ofrak_patch_maker/binary_parser/gnu.py

```diff
@@ -72,6 +63,29 @@ def parse_sections(self, output: str) -> Tuple[Segment, ...]:
             segments.append(seg)
         return tuple(segments)
 
+    def parse_relocations(self, output: str) -> Dict[str, int]:
+        """
+        Use `objdump` with the `--syms` flag to get info on all undefined symbols in a file. Parses
+        columns based on: <https://stackoverflow.com/a/16471895/16690095>.
+        """
+        return self._get_all_symbols(output, False)
+
+    def _get_all_symbols(self, output: str, get_defined: bool) -> Dict[str, int]:
```

Comment:
The name says this gets ALL symbols but one of the required arguments is a flag that (either True or False) means you don't get all the symbols. I would suggest eliminating the `get_defined` flag, then have this helper function return a tuple of name, addr, and symbol_section for each symbol, and let the callers `parse_symbols` and `parse_relocations` filter those appropriately.

Reply:
The data structure holding the tuples doesn't matter, it gets iterated over in its entirety once.